jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -677,13 +677,16 @@ ecma_builtin_helper_def_prop (ecma_object_t *obj_p, /**< object */
   prop_desc.value = value;
 
   prop_desc.is_writable_defined = true;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
   prop_desc.is_writable = writable;
 
   prop_desc.is_enumerable_defined = true;
   prop_desc.is_enumerable = enumerable;
 
   prop_desc.is_configurable_defined = true;
   prop_desc.is_configurable = configurable;
+#pragma GCC diagnostic pop
```

Comment:
The problem is apparent when CONFIG_C99_BOOL8 is not set.
If it is, the build completes successfully, no matter debug or release.

Here you are:

[ 35%] Building C object jerry-core/CMakeFiles/jerry-core.dir/ecma/builtin-objects/ecma-builtin-helpers.c.obj
cd /opt/iotjs/iotjs.git/build/arm-tizenrt/release/deps/jerry/jerry-core && /usr/bin/arm-none-eabi-gcc  -DCONFIG_DISABLE_ES2015_BUILTIN -DCONFIG_DISABLE_ES2015_PROMISE_BUILTIN -DCONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN -DCONFIG_MEM_HEAP_AREA_SIZE=262144 -DJERRY_ENABLE_SNAPSHOT_EXEC -DJERRY_JS_PARSER=1 -DJERRY_NDEBUG -D_BSD_SOURCE -D_DEFAULT_SOURCE -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/debugger -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/base -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/builtin-objects -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/builtin-objects/typedarray -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/operations -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/jcontext -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/jmem -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/jrt -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/lit -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/parser/js -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/parser/regexp -I/opt/iotjs/iotjs.git/deps/jerry/jerry-core/vm -isystem /opt/iotjs/iotjs.git/deps/jerry/jerry-libm/include  -D__TIZENRT__ -Os -fno-strict-aliasing -fno-strength-reduce -fomit-frame-pointer -D__ARM__ -D__arm__ -mthumb -fno-short-enums -mlittle-endian -O2 -mcpu=cortex-r4 -mfpu=vfp3 -DTARGET_BOARD=artik05x -nostdinc -nostdlib --sysroot /opt/iotjs/tizenRT/anew/with_iotjs/tinyara/os -isystem =/include -isystem =/include/tinyara -fno-builtin -fno-stack-protector -Wall -Werror=all -Wextra -Werror=extra -Wformat-nonliteral -Werror=format-nonliteral -Winit-self -Werror=init-self -Wconversion -Werror=conversion -Wsign-conversion -Werror=sign-conversion -Wformat-security -Werror=format-security -Wmissing-declarations -Werror=missing-declarations -Wshadow -Werror=shadow -Wstrict-prototypes -Werror=strict-prototypes -Wundef -Werror=undef -Wold-style-definition -Werror=old-style-definition -Wno-stack-protector -Wno-attributes -Wlogical-op -Werror=logical-op -std=c99 -pedantic -Os -DNDEBUG   -o CMakeFiles/jerry-core.dir/ecma/builtin-objects/ecma-builtin-helpers.c.obj   -c /opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c
/opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c: In function 'ecma_builtin_helper_def_prop':
/opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c:680:27: error: conversion to 'unsigned char:1' from '_Bool8' may alter its value [-Werror=conversion]
   prop_desc.is_writable = writable;
                           ^
/opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c:683:29: error: conversion to 'unsigned char:1' from '_Bool8' may alter its value [-Werror=conversion]
   prop_desc.is_enumerable = enumerable;
                             ^
/opt/iotjs/iotjs.git/deps/jerry/jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c:686:31: error: conversion to 'unsigned char:1' from '_Bool8' may alter its value [-Werror=conversion]
   prop_desc.is_configurable = configurable;
                               ^
cc1: some warnings being treated as errors
jerry-core/CMakeFiles/jerry-core.dir/build.make:833: recipe for target 'jerry-core/CMakeFiles/jerry-core.dir/ecma/builtin-objects/ecma-builtin-helpers.c.obj' failed